Build/Run-Tests.ps1

```diff
@@ -1,20 +1,30 @@
-$testAssemblies = (Get-ChildItem -Path UnitTests -Filter '*Tests.dll' -Recurse -Exclude 'ApprovalTests.dll').FullName | Where-Object { $_.Contains('\bin\Release') }
+$testAssemblies = @();
+$testAssemblies += (Get-ChildItem -Path UnitTests        -Filter '*Tests.dll' -Recurse -Exclude 'ApprovalTests.dll').FullName | Where-Object { $_.Contains('\bin\Release') }
+$testAssemblies += (Get-ChildItem -Path IntegrationTests -Filter '*Tests.dll' -Recurse -Exclude 'ApprovalTests.dll').FullName | Where-Object { $_.Contains('\bin\Release') }
 $packageConfig = [xml](Get-Content .nuget\packages.config)
 $opencover_version = $packageConfig.SelectSingleNode('/packages/package[@id="OpenCover"]').version
 $opencover_console = "packages\OpenCover.$opencover_version\tools\OpenCover.Console.exe"
 
-&$opencover_console `
-    -register:administrator `
-    -returntargetcode `
-    -hideskipped:All `
-    -filter:"+[*]* -[FluentAssertions*]* -[SmartFormat*]* -[nunit*]*" `
-    -excludebyattribute:*.ExcludeFromCodeCoverage* `
-    -excludebyfile:*\*Designer.cs `
-    -output:"OpenCover.GitExtensions.xml" `
-    -target:"nunit3-console.exe" `
-    -targetargs:"$testAssemblies --workers=1 --timeout=90000"
-$testExitCode = $LastExitCode
-Push-AppveyorArtifact "TestResult.xml"
+$testRunCount = 6
+for ($i=1; $i -le $testRunCount; $i++)
+{
+    Write-Host "[INFO]: Test Run ${i}/${testRunCount}"
+    &$opencover_console `
+        -register:administrator `
+        -returntargetcode `
+        -hideskipped:All `
+        -filter:"+[*]* -[FluentAssertions*]* -[SmartFormat*]* -[nunit*]*" `
+        -excludebyattribute:*.ExcludeFromCodeCoverage* `
+        -excludebyfile:*\*Designer.cs `
+        -output:"OpenCover.GitExtensions.xml" `
+        -target:"nunit3-console.exe" `
+        -targetargs:"$testAssemblies --agents=1 --workers=1 --timeout=90000"
+    $testExitCode += $LastExitCode
+    if ($LastExitCode -ne 0) { Write-Host "[ERROR]: Test run ${i} failed!" }
+    $artifact_name = "TestResult" + $i + "." + $LastExitCode + "err.xml"
+    Copy-Item "TestResult.xml" $artifact_name
+    Push-AppveyorArtifact $artifact_name
+}
```

Comment:
Please don't forget to remove this before merging 😉

Reply:
The step back to `$testRunCount = 1` is already waiting to be force-pushed.

At least `-targetargs:"$testAssemblies --agents=1 --workers=1 --timeout=90000"` is essential.

`$artifact_name = "TestResult" + $i + "." + $LastExitCode + "err.xml"` is useful.

Reply:
You can make it an AppVeyor env variable 😉 

And this is what's coming - https://github.com/gitextensions/gitextensions/pull/7396/commits/6efa720c1a0cbe5a6dc4f312e45e6b61ccb8849e